edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -48,62 +47,47 @@ const LectureScreen: ApplicationRoute = () => {
         }
     });
 
-
     // Hooks
     const [numPages, setNumPages] = useState<number>(0);        // Total number of pages in the PDF
     const [page, setPage] = useState<number>(0);                // Current page, starting from 1
     const [uri, setUri] = useState<string>('');                 // Url state
-    //const [isLandscape, setIsLandscape] = useState(false);
-    const [isFullscreen, setIsFullscreen] = useState(false);    // FullScreen display of pdf toggle
+    // UI display setting's hooks
+    const [isLandscape, setIsLandscape] = useState<boolean>(false);       // Landscape display boolean for different UI
+    const [isFullscreen, setIsFullscreen] = useState<boolean>(false);    // FullScreen display of pdf toggle
+    const [widthPercent, setWidthPercent] = useState<string[]>([]);
+    const [heightPercent, setHeightPercent] = useState<string[]>([]);
 
     const [lectureDoc] = usePrefetchedDynamicDoc(CollectionOf<Lecture>(`courses/${courseName}/lectures`), lectureId, undefined);
 
     // Load images when the component mounts
     useEffect(() => {
         if (lectureDoc) {
             getUri();
-            setLandscape();
         }
+    }, [lectureDoc]);
 
-        /* todo for a next sprint, detection of rotation change from landscape to portrait and different UI for each mode
-
-        // Orientation change listener
-        const subscription = ScreenOrientation.addOrientationChangeListener(({ orientationInfo }) => {
-            const { orientation } = orientationInfo;
-            // If the orientation is portrait (either portrait up or down), trigger setPortrait
-            if (
-                orientation === ScreenOrientation.Orientation.PORTRAIT_UP ||
-                orientation === ScreenOrientation.Orientation.PORTRAIT_DOWN
-            ) {
-                setPortrait();
+    useEffect(() => {
+        const onOrientationChange = (currentOrientation: ScreenOrientation.OrientationChangeEvent) => {
+            const orientationValue = currentOrientation.orientationInfo.orientation;
+            if (orientationValue == 1 || orientationValue == 2) {
                 setIsLandscape(false);
             } else {
-                setLandscape();
                 setIsLandscape(true);
             }
-        }); 
+            updateUI();
+        };
 
-        // Cleanup function to remove listener when component unmounts
-        return () => {
-            ScreenOrientation.removeOrientationChangeListener(subscription);
-        };*/
+        const screenOrientationListener =
+            ScreenOrientation.addOrientationChangeListener(onOrientationChange);
 
-    }, [lectureDoc]);
+        return () => {
+            ScreenOrientation.removeOrientationChangeListener(screenOrientationListener);
+        };
+    }, []);
```

Comment:
I have a small concern about not putting an external variable and functions dependancies. The `updateUI` function is being used but is not listed in the dependency array. This could lead un-handled side effect, later if the `updateUI` function is modified in the future. I think that's a good practice to list all external variables and functions that are used inside the useEffect. What do you think about it 🫣?

Reply:
I have went back in more detail on the `usePrefetchedDynamicDoc`. It takes as an argument a `Collection<Type>`. So that's why you needed an array containing a `singleton`. For this reason, you can keep it this way 🦾. In future incremental, we could consider writing another cloud function that tackle this issue in a more reasonable manner 🤟.

Reply:
Yes thank you

Reply:
np! 🐳

Reply:
Yes for the usePrefetchedDynamicDocs we should probably change it in the future, and for the useEffect it is not needed to put a parameter since it is an initialization of a listener that handles the function each time it listens to a change in orientation (I don't know if I understood what you meant exactly)